Tests/SonarScanner.MSBuild.PreProcessor.Test/Roslyn/PluginTests.cs

```diff
@@ -18,77 +18,53 @@
 using SonarScanner.MSBuild.PreProcessor.Roslyn;
 
 namespace SonarScanner.MSBuild.PreProcessor.Test.Roslyn;
 
 [TestClass]
 public class PluginTests
 {
-    [TestMethod]
-    public void Ctor_WhenKeyIsInvalid_Throws()
+    public void Plugin_IsValid()
     {
-        // 1. Null
-        Action act = () => new Plugin(null, "v1", "resourceName");
-        act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("key");
-
-        // 2. Empty
-        act = () => new Plugin(string.Empty, "v1", "resourceName");
-        act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("key");
-
-        // 3. Whitespace
-        act = () => new Plugin("\r ", "v1", "resourceName");
-        act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("key");
+        var plugin = new Plugin() { Key = "pluginKey", Version = "42", StaticResourceName = "test.zip" };
+        plugin.IsValid.Should().BeTrue();
     }
 
-    [TestMethod]
-    public void Ctor_WhenVersionIsInvalid_Throws()
+    [DataTestMethod]
+    [DataRow("pluginKey", "42", null)]
+    [DataRow("pluginKey", null, "test.zip")]
+    [DataRow(null, "42", "test.zip")]
+    [DataRow("pluginKey", null, null)]
+    [DataRow(null, "42", null)]
+    [DataRow(null, null, "test.zip")]
+    [DataRow(null, null, null)]
+    public void Plugin_Is_InValid(string key, string version, string resourceName)
     {
-        // 1. Null
-        Action act = () => new Plugin("key", null, "resourceName");
-        act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("version");
-
-        // 2. Empty
-        act = () => new Plugin("key", string.Empty, "resourceName");
-        act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("version");
-
-        // 3. Whitespace
-        act = () => new Plugin("key", "\r\n ", "resourceName");
-        act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("version");
+        var plugin = new Plugin() { Key = key, Version = version, StaticResourceName = resourceName };
+        plugin.IsValid.Should().BeFalse();
     }
 
     [TestMethod]
-    public void Ctor_WhenStaticResourceNameIsNull_Throws()
+    public void Plugin_AddProperty_Populates_Correctly()
     {
-        // 1. Null
-        Action act = () => new Plugin("key", "version", null);
-        act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("staticResourceName");
-
-        // 2. Empty
-        act = () => new Plugin("key", "version", string.Empty);
-        act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("staticResourceName");
-
-        // 3. Whitespace
-        act = () => new Plugin("key", "version", "\r\n ");
-        act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("staticResourceName");
+        var plugin = new Plugin();
+        plugin.AddProperty("pluginKey", "someKey");
```

Comment:
Separate scaffolding from assertions to follow our usual test shape. Write 4x `AddProperty` and then 3x `Should()`

The last `AddProperty` should be the `("someOtherProperty", "someOtherValue")` as before, to prove that it's a NO-OP